File: JavaScript/SortingAlgorithms/InsertionSort.js
function insertionSort (arr) {

    for (let i = 1; i < arr.length; i++) {
        let temp = arr[i];
        let idx = i - 1;
        while (arr[idx] > temp && idx >= 0) {
            arr[idx + 1] = arr[idx];
            idx--;
        }
        arr[idx + 1] = temp;
        console.log(arr);
    }

    return arr;
}

/**
* Uncomment the following lines to test
*/
// console.log(insertionSort([5, 8, 2, 0, 1]));
// console.log(insertionSort([4, 3, 2, 10, 12, 1, 5, 6]));

File: Python/SortingAlgorithms/InsertionSort.py
def insertionSort(arr):

    for i in range(1, len(arr)):
        temp = arr[i]
        idx = i

        while arr[idx - 1] > temp and idx > 0:
            arr[idx] = arr[idx - 1]
            idx = idx - 1

        arr[idx] = temp

    return arr


# Uncomment the following lines to test
# print(insertionSort([5, 8, 2, 0, 1]))
# print(insertionSort([4, 3, 2, 10, 12, 1, 5, 6]))
